example/fly_mission/fly_mission.cpp

```diff
@@ -0,0 +1,264 @@
+        }
+
+        future_result.get();
+    }
+
+    dc.register_on_timeout([](uint64_t uuid) {
+        std::cout << "Device with UUID timed out: " << uuid << std::endl;
+        std::cout << "Exiting." << std::endl;
+        exit(0);
+    });
+
+    // We don't need to specifiy the UUID if it's only one device anyway.
+    // If there were multiple, we could specify it with:
+    // dc.device(uint64_t uuid);
+    Device &device = dc.device();
+
+    while (!device.telemetry().health_all_ok()) {
+        std::cout << "Waiting for device to be ready" << std::endl;
+        std::this_thread::sleep_for(std::chrono::seconds(1));
+    }
+
+    std::cout << "Device ready" << std::endl;
+    std::cout << "Creating and uploading mission" << std::endl;
+
+    std::vector<std::shared_ptr<MissionItem>> mission_items;
+
+    mission_items.push_back(
+        add_mission_item(47.398170327054473,
+                         8.5456490218639658,
+                         10.0f, 5.0f, false,
+                         20.0f, 60.0f,
+                         MissionItem::CameraAction::NONE));
+
+    mission_items.push_back(
+        add_mission_item(47.398241338125118,
+                         8.5455360114574432,
+                         10.0f, 2.0f, true,
+                         0.0f, -60.0f,
+                         MissionItem::CameraAction::TAKE_PHOTO));
+
+    mission_items.push_back(
+        add_mission_item(47.398139363821485, 8.5453846156597137,
+                         10.0f, 5.0f, true,
+                         -46.0f, 0.0f,
+                         MissionItem::CameraAction::START_VIDEO));
+
+    mission_items.push_back(
+        add_mission_item(47.398058617228855,
+                         8.5454618036746979,
+                         10.0f, 2.0f, false,
+                         -90.0f, 30.0f,
+                         MissionItem::CameraAction::STOP_VIDEO));
+
+    mission_items.push_back(
+        add_mission_item(47.398100366082858,
+                         8.5456969141960144,
+                         10.0f, 5.0f, false,
+                         -45.0f, -30.0f,
+                         MissionItem::CameraAction::START_PHOTO_INTERVAL));
+
+    mission_items.push_back(
+        add_mission_item(47.398001890458097,
+                         8.5455576181411743,
+                         10.0f, 5.0f, false,
+                         0.0f, 0.0f,
+                         MissionItem::CameraAction::STOP_PHOTO_INTERVAL));
+
+    {
+        std::cout << "Uploading mission..." << std::endl;
+        // We only have the send_mission function asynchronous for now, so we wrap it using
+        // std::future.
+        auto prom = std::make_shared<std::promise<Mission::Result>>();
+        auto future_result = prom->get_future();
+        device.mission().send_mission_async(
+        mission_items, [prom](Mission::Result result) {
+            prom->set_value(result);
+            std::cout << "Mission uploaded." << std::endl;
```

Comment:
@julianoes I think at this point we don't know if the mission uploaded. This should be "Mission uploa**ding**"

Reply:
You're right, I moved it in c67e45c.